app/src/main/java/org/kiwix/kiwixmobile/nav/destination/reader/ReaderFragment.kt

```diff
@@ -52,30 +57,91 @@ import org.kiwix.kiwixmobile.core.utils.UpdateUtils
 import org.kiwix.kiwixmobile.kiwixActivityComponent
 import org.kiwix.kiwixmobile.main.KiwixWebViewClient
+import org.kiwix.kiwixmobile.navigate
 import org.kiwix.kiwixmobile.webserver.ZimHostActivity
 import org.kiwix.kiwixmobile.zim_manager.ZimManageActivity
 import java.io.File
 
+private const val HIDE_TAB_SWITCHER_DELAY: Long = 300
+
 class ReaderFragment : CoreReaderFragment() {
+  private val args: ReaderFragmentArgs by navArgs()
+
   override fun inject(baseActivity: BaseActivity) {
     baseActivity.kiwixActivityComponent.inject(this)
   }
 
-  private val args: ReaderFragmentArgs by navArgs()
-
   override fun onActivityCreated(savedInstanceState: Bundle?) {
     super.onActivityCreated(savedInstanceState)
-    manageExternalLaunchAndRestoringViewState()
-    val uri = args.zimFileUri
+    manageExternalLaunchAndRestoringViewState(args.zimFileUri)
 
-    if (uri.isNotEmpty()) {
-      openZimFile(Uri.parse(uri).toFile())
+    noOpenBookButton.setOnClickListener {
+      (activity as AppCompatActivity).navigate(
+        ReaderFragmentDirections.actionNavigationReaderToNavigationLibrary()
+      )
+    }
+  }
+
+  override fun showHomePage() {
+    exitBook()
+  }
+
+  private fun exitBook() {
+    // getCurrentWebView().removeAllViews()
+    showNoBookOpenViews()
+    bottomToolbar.visibility = GONE
+    actionBar.title = getString(R.string.reader)
+    contentFrame.visibility = GONE
+    mainMenu?.hideBookSpecificMenuItems()
+    // close book
+    zimReaderContainer.setZimFile(null)
+    mainMenu?.hideBookSpecificMenuItems()
+  }
+
+  override fun openHomeScreen() {
+    Handler().postDelayed({
+      if (webViewList.size == 0) {
+        hideTabSwitcher()
+      }
+    }, HIDE_TAB_SWITCHER_DELAY)
+  }
+
+  override fun hideTabSwitcher() {
+    if (actionBar != null) {
+      actionBar.setDisplayHomeAsUpEnabled(false)
+      actionBar.setDisplayShowTitleEnabled(true)
+
+      setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED)
+      closeAllTabsButton.setImageDrawable(
+        ContextCompat.getDrawable(
+          requireActivity(),
+          org.kiwix.kiwixmobile.core.R.drawable.ic_close_black_24dp
+        )
+      )
+      if (tabSwitcherRoot.visibility == View.VISIBLE) {
+        tabSwitcherRoot.visibility = GONE
+        startAnimation(tabSwitcherRoot, org.kiwix.kiwixmobile.core.R.anim.slide_up)
+        progressBar.visibility = View.VISIBLE
+        progressBar.progress = 0
+        contentFrame.visibility = View.VISIBLE
+      }
+      if (mainMenu != null) {
+        mainMenu.showWebViewOptions(true)
+      }
+      if (webViewList.isEmpty()) {
+        exitBook()
+      } else {
+        selectTab(currentWebViewIndex)
+      }
     }
   }
 
   override fun onCreateOptionsMenu(menu: Menu, menuInflater: MenuInflater) {
     super.onCreateOptionsMenu(menu, menuInflater)
     menu.findItem(R.id.menu_new_navigation)?.isVisible = false
+    if (zimReaderContainer?.zimFileReader == null) {
```

Comment:
zimReaderContainer shouldn't be nullable

Reply:
good finds! thanks.